src/pkg/k8s/pods.go

```diff
@@ -79,13 +79,15 @@ func (k *K8s) CreatePod(pod *corev1.Pod) (*corev1.Pod, error) {
 
 // GetAllPods returns a list of pods from the cluster for all namespaces.
 func (k *K8s) GetAllPods() (*corev1.PodList, error) {
-	return k.GetPods(corev1.NamespaceAll)
+	return k.GetPods(corev1.NamespaceAll, nil)
 }
 
 // GetPods returns a list of pods from the cluster by namespace.
-func (k *K8s) GetPods(namespace string) (*corev1.PodList, error) {
-	metaOptions := metav1.ListOptions{}
-	return k.Clientset.CoreV1().Pods(namespace).List(context.TODO(), metaOptions)
+func (k *K8s) GetPods(namespace string, listOpts *metav1.ListOptions) (*corev1.PodList, error) {
+	if listOpts == nil {
+		listOpts = &metav1.ListOptions{}
+	}
+	return k.Clientset.CoreV1().Pods(namespace).List(context.TODO(), *listOpts)
```

Comment:
Should not mutate types from non-pointer to pointer when not needed: `List(ctx context.Context, opts metav1.ListOptions)`

Change this from `listOpts *metav1.ListOptions` to `listOpts metav1.ListOptions`, and instead of passing in `nil` when blank, use `metav1.ListOptions{}`

Reply:
updated